### PR TITLE
Psl 305 update gonode to match dd service

### PR DESCRIPTION
**Pull request by Bjbrostoff**
PSL-305 brings gonode in line with updates to the dupe detection service.
1) Adapt protobuf to fit new models for rareness score grpc comms
2) Make corresponding changes to the Fingerprints struct
3) Make corresponding changes to the API
4) Adapt walletnode's nft registration and sense registration services to use the new API metadata
5) Adapt supernode's nft registration and sense registration services to absorb the new API metadata 
6) Adapt the dupe detection service and handler to work with the new struct models
7) Adapt dupe detection scan service to properly find and store new fingerprints based on discovered new nft registration and sense tickets.  This included creating a new pastel client api call to list action tickets.

### CODE REVIEW

**Comment by Bjbrostoff** [comment]
Forgot to add some code in QR generation.

**Comment by Bjbrostoff** [comment]
Reopen code for intermediate pull instead

**Comment by a-ok123** [comment]
You will remove all that commented out code, right?

**Review by a-ok123 — DISMISSED** [review]

File: dupedetection/cmd/test.go [hunk 1]
```diff
@@ -27,13 +27,13 @@ func main() {
...
+	res, err := client.ImageRarenessScore(context.Background(), data, "png", "testBlockHash", "240669", "2022-03-31 16:55:28", "testPastelID", "jXYiHNqO9B7psxFQZb1thEgDNykZjL8GkHMZNPZx3iCYre1j3g0zHynlTQ9TdvY6dcRlYIsNfwIQ6nVXBSVJis", "jXpDb5K6S81ghCusMOXLP6k0RvqgFhkBJSFf6OhjEmpvCWGZiptRyRgfQ9cTD709sA58m5czpipFnvpoHuPX0F", "jXS9NIXHj8pd9mLNsP2uKgIh1b3EH2aq5dwupUF7hoaltTE8Zlf6R7Pke0cGr071kxYxqXHQmfVO5dA4jH0ejQ", false, "")
 	if err != nil {
 		panic("do ImageRarenessScore() error" + err.Error())
 	}
 
-	dataJson, _ := json.Marshal(res)
+	dataJSON, _ := json.Marshal(res)
 
-	fmt.Println(string(dataJson))
+	fmt.Println(string(dataJSON))
```
Comment (mateeullahmalik) [inline dupedetection/cmd/test.go:37]: i think we can remove println

Comment (mateeullahmalik) [inline dupedetection/cmd/test.go:37]: i think we can remove println

File: dupedetection/ddclient/ddclient.go [hunk 2]
```diff
@@ -78,101 +85,42 @@ func (ddClient *ddServerClientImpl) callImageRarenessScore(ctx context.Context,
 		return nil, errors.Errorf("send request: %w", err)
 	}
 
-	/*
```
Comment (mateeullahmalik) [inline dupedetection/ddclient/ddclient.go:81]: can we remove this pls?

Comment (mateeullahmalik) [inline dupedetection/ddclient/ddclient.go:81]: can we remove this pls?

File: dupedetection/ddscan/service.go [hunk 3]
```diff
@@ -284,38 +257,79 @@ func (s *service) storeFingerprint(ctx context.Context, input *dupeDetectionFing
...
+		return nil, errors.Errorf("decode data: %w", err)
+	}
+
+	decData, err := zstd.Decompress(nil, dec)
+	if err != nil {
+		return nil, errors.Errorf("decompress: %w", err)
+	}
+
+	splits := bytes.Split(decData, []byte{pastel.SeparatorByte})
+	file, err := utils.B64Decode(splits[0])
```
Comment (mateeullahmalik) [inline dupedetection/ddscan/service.go:321]: check length first to avoid crash?

Comment (mateeullahmalik) [inline dupedetection/ddscan/service.go:321]: check length first to avoid crash?

**Review by mateeullahmalik — DISMISSED** [review]

File: dupedetection/ddscan/service.go (hunk 3 above)
Comment (Bjbrostoff) [reply dupedetection/ddscan/service.go:321]: Good idea

**Review by mateeullahmalik — APPROVED** [review]

**Pull request merged**